### PR TITLE
Feat: search

### PR DESCRIPTION
**Pull request by ingvaar**
Add some research capability using [easyops-cn/docusaurus-search-local](https://github.com/easyops-cn/docusaurus-search-local) plugin.

Search only works with statically built documentation (after a `yarn build`) and can be tested by using `yarn run serve` (after build). It won't work in development (running a `yarn start`).

### CODE REVIEW

**Review by ccamel — APPROVED** [review]
A must have. Thanks 👍

File: src/scss/custom.scss [hunk 1]
```diff
@@ -90,3 +90,8 @@ html {
 [data-theme="dark"] .docusaurus-highlight-code-line {
   background-color: rgb(0 0 0 / 30%);
 }
+
+html[data-theme="dark"] {
```
Comment (fredericvilcot) [inline src/scss/custom.scss:94]: This is useless as main variables are already injected into `html` element, so you can easily append new variables in dedicated [selector](https://github.com/okp4/docs/blob/0050e91aea8efa1a9fba42dc12c3821d9c0806ae/src/scss/custom.scss#L40)

File: src/scss/custom.scss [hunk 2]
```diff
@@ -90,3 +90,8 @@ html {
...
+  --search-local-highlight-color: #2d66c3;
```
Comment (fredericvilcot) [inline src/scss/custom.scss:95]: Maybe we can keep on following OKP4 palette by using here main blue hexadecimal value:
```suggestion
  --search-local-highlight-color: #0f1e3d;
```

**Review by fredericvilcot — CHANGES_REQUESTED** [review]
Good job, just some minor remarks 😎

**Pull request closed without merging**